src/server/page_server.cpp

```diff
@@ -99,6 +103,20 @@ page_server::receive_log_page_fetch (cubpacking::unpacker &upk)
     }
 }
 
+void
+page_server::receive_data_page_fetch (cubpacking::unpacker &upk)
+{
+  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_DATA_PAGE))
+    {
+        VPID vpID;
+        std::string message;
+
+        upk.unpack_string (message);
+        std::memcpy (&vpID.pageid, message.c_str (), sizeof (vpID.pageid));
+        std::memcpy (&vpID.volid, message.c_str () + sizeof (vpID.pageid), sizeof (vpID.volid));

```

Comment:
What do you think about copying the whole VPID at once? Sending request would have to be updated too.
```suggestion
        std::memcpy (&vpID, message.c_str (), sizeof (vpID));
```